rviz_default_plugins/src/rviz_default_plugins/displays/effort/effort_display.cpp

```diff
@@ -0,0 +1,416 @@
+ *
+ *     * Redistributions of source code must retain the above copyright
+ *       notice, this list of conditions and the following disclaimer.
+ *     * Redistributions in binary form must reproduce the above copyright
+ *       notice, this list of conditions and the following disclaimer in the
+ *       documentation and/or other materials provided with the distribution.
+ *     * Neither the name of the Willow Garage, Inc. nor the names of its
+ *       contributors may be used to endorse or promote products derived from
+ *       this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "rviz_default_plugins/displays/effort/effort_display.hpp"
+
+#include <OgreSceneNode.h>
+#include <OgreSceneManager.h>
+#include <urdf/model.h>
+
+#include <QString>
+
+#include <chrono>
+#include <cstddef>
+#include <memory>
+#include <string>
+#include <vector>
+
+#include <rviz_common/validate_floats.hpp>
+
+#include <rviz_common/properties/property.hpp>
+#include <rviz_rendering/objects/effort_visual.hpp>
+
+using namespace std::chrono_literals;
+
+namespace rviz_default_plugins
+{
+namespace displays
+{
+JointInfo::JointInfo(const std::string & name, rviz_common::properties::Property * parent_category)
+{
+  name_ = name;
+  effort_ = 0;
+  max_effort_ = 0;
+
+  // info->category_ = new Property( QString::fromStdString( info->name_ ) , QVariant(), "",
+  // joints_category_);
+  category_ = new rviz_common::properties::Property(
+    QString::fromStdString(name_), true, "", parent_category,
+    SLOT(updateVisibility()), this);
+
+  effort_property_ = new rviz_common::properties::FloatProperty(
+    "Effort", 0, "Effort value of this joint.", category_);
+  effort_property_->setReadOnly(true);
+
+  max_effort_property_ =
+    new rviz_common::properties::FloatProperty(
+    "Max Effort", 0, "Max Effort value of this joint.", category_);
+  max_effort_property_->setReadOnly(true);
+}
+
+JointInfo::~JointInfo()
+{
+}
+
+JointInfo * EffortDisplay::createJoint(const std::string & joint)
+{
+  JointInfo * info = new JointInfo(joint, joints_category_);
+  joints_.insert(std::make_pair(joint, info));
```

Comment:
I think you need `#include <utility>` at the top of the file for `std::make_pair`.

More importantly, what is responsible for freeing the `info` pointer here?